### PR TITLE
add dinamic list of goods solution

### PR DESCRIPTION
**Pull request by opiquem**
[DEMO LINK](https://maksym-bobryk.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/api/goods.ts [hunk 1]
```diff
@@ -5,15 +5,24 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
...
+    .then(goods => goods
+      .sort((g1, g2) => g1.name.localeCompare(g2.name))
+      .slice(0, 5));
 };
 
 export const getRedGoods = () => {
   return getAll()
-    .then(goods => goods); // get only red
+    .then(goods => goods
+      .filter(good => good.color === 'red'));
```
Comment (33Orange) [inline src/api/goods.ts:27]:
```suggestion
    .then(goods => goods.filter(good => good.color === 'red'));
```

File: src/GoodsList.tsx [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
-  <ul>
-    {goods.map(good => (
-      <li key={good.id} data-cy="good">
-        {good.name}
-      </li>
-    ))}
-  </ul>
-);
+export const GoodsList: React.FC<Props> = React.memo(({ goods }) => {
+  const classes = (good: Good) => (
```
Comment (33Orange) [inline src/GoodsList.tsx:11]:
```suggestion
  const getClasses = (good: Good) => (
```

File: src/GoodsList.tsx [hunk 3]
```diff
@@ -1,16 +1,67 @@
...
+    cn(
+      {
+        'has-text-link': good.color === 'blue',
+        'has-text-danger': good.color === 'red',
+        'has-text-success': good.color === 'green',
+      },
+    ));
```
Comment (33Orange) [inline src/GoodsList.tsx:18]: Move this function outside your component to avoid re-creating every render.

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job.

A few moments to improve

File: src/api/goods.ts [hunk 4]
```diff
@@ -5,15 +5,23 @@ const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/good
...
+
+      return response.json();
+    });
 }
 
 export const get5First = () => {
   return getAll()
-    .then(goods => goods); // sort and get the first 5
+    .then(goods => goods
+      .sort((g1, g2) => g1.name.localeCompare(g2.name))
```
Comment (dffUqp) [inline src/api/goods.ts:20]: bad practice to use a number in variable naming, I think we can use `a` and `b` in sort method.
It is normal so don't worry(but only for sort method)

**Review by dffUqp — APPROVED** [review]
GJ✔️

**Pull request closed without merging**